crates/router/src/core/pm_auth.rs

```diff
@@ -508,22 +490,16 @@ async fn store_bank_details_in_payment_methods(
             .change_context(ApiErrorResponse::InternalServerError)
             .attach_printable("Unable to encrypt customer details")?;
 
-            #[cfg(all(
-                any(feature = "v1", feature = "v2"),
-                not(feature = "payment_methods_v2")
-            ))]
+            #[cfg(feature = "v1")]
             let pm_id = generate_id(consts::ID_LENGTH, "pm");
 
-            #[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+            #[cfg(feature = "v2")]
             let pm_id = common_utils::id_type::GlobalPaymentMethodId::generate("random_cell_id")
                 .change_context(errors::ApiErrorResponse::InternalServerError)
                 .attach_printable("Unable to generate GlobalPaymentMethodId")?;
 
             let now = common_utils::date_time::now();
-            #[cfg(all(
-                any(feature = "v1", feature = "v2"),
-                not(feature = "payment_methods_v2")
-            ))]
+            #[cfg(feature = "v1")]
```

Comment:
This code also seems to be part of the same function `store_bank_details_in_payment_methods()`, which is behind v1 feature flag.